controllers/managedclusterview_controller.go

```diff
@@ -0,0 +1,186 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	viewv1beta1 "github.com/stolostron/multicloud-operators-foundation/pkg/apis/view/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type ManagedClusterViewReconciler struct {
+	Client client.Client
+	Logger *slog.Logger
+}
+
+const (
+	ODFInfoConfigMapName    = "odf-info"
+	ConfigMapResourceType   = "ConfigMap"
+	ClientInfoConfigMapName = "odf-client-info"
+)
+
+// StorageClusterConfig represents the structure of the storage cluster configuration.
+type StorageClusterConfig struct {
+	Clients        []string `json:"Clients,omitempty"`
+	DeploymentType string   `json:"DeploymentType"`
+	StorageCluster struct {
+		CephClusterFSID string `json:"CephClusterFSID"`
+		NamespacedName  struct {
+			Name      string `json:"Name"`
+			Namespace string `json:"Namespace"`
+		} `json:"NamespacedName"`
+		StorageProviderEndpoint string `json:"StorageProviderEndpoint"`
+	} `json:"StorageCluster"`
+	StorageSystemName   string `json:"StorageSystemName"`
+	Version             string `json:"Version"`
+	ProviderClusterName string `json:"ProviderClusterName,omitempty"`
+}
```

Comment:
If we don't use the same json tags as [here](https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/odfinfoconfig.go#L23-L50) marshall will fail right?

Reply:
I will need to wait for it to be exported in OCS. Once that is done , i will update it to use that but you can still review the basic functions and tests.